adapters/src/main/kotlin/com/krzykrucz/elesson/currentlesson/preparedness/adapters/findcurrentlesson/CurrentLessonDatabaseAdapter.kt

```diff
@@ -0,0 +1,36 @@
+package com.krzykrucz.elesson.currentlesson.preparedness.adapters.findcurrentlesson
+
+import arrow.core.toOption
+import arrow.effects.IO
+import com.krzykrucz.elesson.currentlesson.attendance.domain.CheckedAttendanceList
+import com.krzykrucz.elesson.currentlesson.monolith.Database
+import com.krzykrucz.elesson.currentlesson.preparedness.domain.FindCurrentLesson
+import com.krzykrucz.elesson.currentlesson.preparedness.domain.UnpreparednessError
+import com.krzykrucz.elesson.currentlesson.shared.CurrentLesson
+import com.krzykrucz.elesson.currentlesson.shared.LessonAfterAttendance
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class AdapterConfig {
+    @Bean
+    fun getLessonBean(): FindCurrentLesson = {
+        Database.LESSON_DATABASE[it].toOption()
+            .toEither { UnpreparednessError.LessonNotStarted }
+            .map { persistentCurrentLesson ->
+
+                (if (persistentCurrentLesson.lessonTopic == null &&
+                    persistentCurrentLesson.attendance is CheckedAttendanceList) {
+                    LessonAfterAttendance(
+                        persistentCurrentLesson.lessonId,
+                        persistentCurrentLesson.attendance as CheckedAttendanceList,
+                        persistentCurrentLesson.unpreparedStudents
+                    )
+                } else {
+                    null//TODO
```

Comment:
`//TODO` is left here, when will it be implemented? returning `null` doesn't look safe

Reply:
yes, I forgot to move all lesson types here
I'm leaning now towards keeping a unified model instead of duplicating e.g. a PresentStudent concept between certain pipelines, because PresentStudent means always the same

thus, I will probably move all lesson stages to implement to CurrentLesson sealed class and I will add them to this implementation

Reply:
this has its drawback, though, because it introduces some dependencies between pipelines, which could have been avoided